cou/apps/base.py

```diff
@@ -516,6 +523,33 @@ def _get_upgrade_current_release_packages_step(
 
         return step
 
+    def _get_charm_hub_migration(self) -> PreUpgradeStep:
+        if not self.is_from_charm_store:
+            return PreUpgradeStep()
+
+        return PreUpgradeStep(
+            f"Migrate '{self.name}' from charmstore to charmhub",
+            coro=self.model.upgrade_charm(
+                self.name, self.possible_current_channel, switch=f"ch:{self.charm}"
+            ),
+        )
+
+    def _get_change_to_openstack_channels(self) -> PreUpgradeStep:
+        if self.channel != LATEST_STABLE:
+            return PreUpgradeStep()
+        return PreUpgradeStep(
+            f"Change '{self.name}' channel from {self.channel} to {self.possible_current_channel}. "
+            "COU cannot guarantee that this is not a downgrade",
+            coro=self.model.upgrade_charm(self.name, self.possible_current_channel),
+        )
+
+    def _need_to_refresh(self, target) -> bool:
+        return self.can_upgrade_to or self.channel_codename < target
+
+    @property
+    def _can_refresh(self) -> bool:
```

Comment:
Why doing "private" property here? And please try to keep all properties in one place.

Reply:
Yeah, I think it doesn't make sense to have private property. I've changes and put it next to the other properties